src/main/java/org/red5/server/net/rtmp/RTMPConnection.java

```diff
@@ -673,7 +673,7 @@ public boolean isValidStreamId(Number streamId) {
 		if (log.isTraceEnabled()) {
 			log.trace("Checking validation for streamId {}; reservedStreams: {}; streams: {}, connection: {}", new Object[] { d, reservedStreams, streams, sessionId });
 		}
-		if (d < 0 || !reservedStreams.contains(d)) {
+		if (d <= 0 || !reservedStreams.contains(d)) {

```

Comment:
Ensure that the equals does not cause issues with a unit test, I think there was a reason for not allowing 0 here.